asllib/ASTUtils.ml

```diff
@@ -506,12 +512,24 @@ exception FailedConstraintOp
 
 let constraint_binop op =
   let do_op c1 c2 =
-    match (c1, c2) with
-    | Constraint_Exact e1, Constraint_Exact e2 ->
+    match (c1, c2, op) with
```

Comment:
nit: you can first match the a MINUS op and use it to reverse a c2 range constraint. Then, you only have to handle the PLUS case.